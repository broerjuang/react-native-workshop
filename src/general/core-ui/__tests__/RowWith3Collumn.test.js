import React from 'react';
import renderer from 'react-test-renderer';
import RowWith3Collumn from '../RowWith3Collumn';
import {View} from 'react-native';

describe('RowWith3Collumn', () => {
  let edge = <View style={{flex: 1, backgroundColor: 'blue'}} />;
  let middle = <View style={{flex: 1, backgroundColor: 'red'}} />;
  it('should render RowWith3Collumn corectly', () => {
    let component = renderer.create(
      <RowWith3Collumn left={edge} middle={middle} right={edge} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
