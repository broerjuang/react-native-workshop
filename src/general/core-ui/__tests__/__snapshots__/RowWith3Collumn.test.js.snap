// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowWith3Collumn should render RowWith3Collumn corectly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "15%",
          },
          Object {
            "height": 30,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "blue",
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
            "padding": 10,
            "width": "70%",
          },
          Object {
            "height": 30,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "15%",
          },
          Object {
            "height": 30,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "blue",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
